lib/std/net.zig

```diff
@@ -636,17 +636,35 @@ pub fn connectUnixSocket(path: []const u8) !Stream {
 }
 
 fn if_nametoindex(name: []const u8) !u32 {
-    var ifr: os.ifreq = undefined;
-    var sockfd = try os.socket(os.AF.UNIX, os.SOCK.DGRAM | os.SOCK.CLOEXEC, 0);
-    defer os.closeSocket(sockfd);
+    if (builtin.target.os.tag == .linux) {
+        var ifr: os.ifreq = undefined;
+        var sockfd = try os.socket(os.AF.UNIX, os.SOCK.DGRAM | os.SOCK.CLOEXEC, 0);
+        defer os.closeSocket(sockfd);
+
+        std.mem.copy(u8, &ifr.ifrn.name, name);
+        ifr.ifrn.name[name.len] = 0;
+
+        // TODO investigate if this needs to be integrated with evented I/O.
+        try os.ioctl_SIOCGIFINDEX(sockfd, &ifr);
 
-    std.mem.copy(u8, &ifr.ifrn.name, name);
-    ifr.ifrn.name[name.len] = 0;
+        return @bitCast(u32, ifr.ifru.ivalue);
+    }
+
+    if (builtin.target.os.tag == .macos) {
```

Comment:
This could also check for any of `.macos`, `.ios`, `.tvos`, `.watchos`, as is done [here](https://github.com/ziglang/zig/blob/master/lib/std/c.zig#L45) if that is preferable.

Reply:
How about `builtin.target.isDarwin()` then?

Reply:
Perfect, I wasn't aware of that function! Will update 👍

Reply:
@kubkon updated!